src/integrationTest/java/uk/gov/hmcts/darts/common/controller/CourthouseApiTest.java

```diff
@@ -0,0 +1,179 @@
+package uk.gov.hmcts.darts.common.controller;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.JsonPath;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import uk.gov.hmcts.darts.courthouse.CourthouseRepository;
+import uk.gov.hmcts.darts.courthouse.model.ExtendedCourthouse;
+import uk.gov.hmcts.darts.courthouse.service.CourthouseService;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.darts.common.util.TestUtils.getContentsFromFile;
+
+@SpringBootTest
+@ActiveProfiles({"intTest", "h2db"})
+@AutoConfigureMockMvc
+@SuppressWarnings({"PMD.JUnitTestsShouldIncludeAssert","PMD.AvoidDuplicateLiterals"})
+class CourthouseApiTest {
+
+    public static final String REQUEST_BODY_HAVERFORDWEST_JSON = "Tests/CourthousesTest/courthousesPostEndpoint/requestBodyHaverfordwest.json";
+    private static final String REQUEST_BODY_SWANSEA_JSON = "Tests/CourthousesTest/courthousesPostEndpoint/requestBodySwansea.json";
+    public static final String REQUEST_BODY_400_MISSING_COURTHOUSE_NAME_JSON = "Tests/CourthousesTest/courthousesPostEndpoint/requestBody400_MissingCourthouseName.json";
+    @Autowired
+    private CourthouseService courthouseService;
+
+    @Autowired
+    private CourthouseRepository courthouseRepository;
+
+    @Autowired
+    private transient MockMvc mockMvc;
+
+
+    @BeforeEach
+    void setUp() {
+        courthouseRepository.deleteAll();
```

Comment:
Do we need to do a deleteAll here - wondering if there is a chance that somebody will accidentally delete an important table from a "good" postgres database

Reply:
So the integration tests are using an in memory database and this will delete all records in between tests. Im not sure how it will be possible to delete the postgres database.